src/api/OnmsServer.ts

```diff
@@ -82,6 +82,18 @@ export class OnmsServer {
     return uri.toString();
   }
 
+  /**
+   * Whether this server object is the same as another.
+   */
+  public equals(that?: OnmsServer) {
+    return that &&
+      this.id === that.id &&
+      this.name === that.name &&
+      this.url === that.url &&
+      (this.auth === that.auth || (this.auth && this.auth.equals(that.auth))) &&
+      (this.metadata === that.metadata || (this.metadata && this.metadata.equals(that.metadata)));
```

Comment:
same as above, I like the `&&`/`||` to be on the next line due to easier readability.